bin/mocha.js

```diff
@@ -109,7 +113,15 @@ if (mochaArgs['node-option'] || Object.keys(nodeArgs).length || hasInspect) {
   proc.on('exit', (code, signal) => {
     process.on('exit', () => {
       if (signal) {
-        process.kill(process.pid, signal);
+        const numericSignal =
+          typeof signal === 'string' ? os.constants.signals[signal] : signal;
+        if (mochaArgs['posix-exit-codes'] === true) {
+          process.exit(SIGNAL_OFFSET + numericSignal);
+        } else {
+          process.kill(process.pid, signal);
+        }
+      } else if (code !== 0 && mochaArgs['posix-exit-codes'] === true) {
+        process.exit(EXIT_FAILURE);
       } else {
         process.exit(code);
```

Comment:
Could probably make it this:

```suggestion
      } else {
        process.exit(Math.min(code, mochaArgs['posix-exit-codes'] ? 1 : 255));
```

That reuses the `Math.min(code, 255)` logic that's in other places.

Not sure the `EXIT_FAILURE` helps much here, the direct `1` makes it possible to use the `Math.min` like this